src/utils/calculateReClammInitAmounts.ts

```diff
@@ -0,0 +1,59 @@
+import {
+    type InputAmount,
+    type PoolState,
+    type PublicWalletClient,
+    fpMulDown,
+    fpDivDown,
+    isSameAddress,
+} from 'src';
+import { parseAbi } from 'viem';
+
+export async function calculateReClammInitAmounts({
+    client,
+    poolState,
+    givenAmountIn,
+}: {
+    client: PublicWalletClient;
+    poolState: PoolState;
+    givenAmountIn: InputAmount;
+}): Promise<InputAmount[]> {
+    // fetch proportion value from pool contract
+    const proportion = await client.readContract({
+        address: poolState.address,
+        abi: parseAbi([
+            'function computeInitialBalanceRatio() external view returns (uint256 balanceRatio)',
+        ]),
+        functionName: 'computeInitialBalanceRatio',
+        args: [],
+    });
```

Comment:
The way this is currently implemented requires pool to already be deployed, which would wreck the pool creation UX, which has user enter token amounts before deploying the pool

Reply:
I'm hoping it is possible to recreate logic off chain, otherwise I gotta take a sledge hammer to the creation UX

Reply:
If we're already replicating math off-chain on balancer-maths, I'd rather source it from there as well 👍

Reply:
The maths have been pushed to [johns PR](https://github.com/balancer/balancer-maths/pull/101)

After the release, I will replace this on chain fetch